pytorch_lightning/utilities/optimizer.py

```diff
@@ -29,5 +29,7 @@ def optimizers_to_device(optimizers: Iterable[Optimizer], device: _DEVICE) -> No
 
 def optimizer_to_device(optimizer: Optimizer, device: _DEVICE) -> None:
     """Moves the state of a single optimizer to the device."""
-    for p, v in optimizer.state.items():
-        optimizer.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, device)
+    state, param_groups = optimizer.state_dict().values()
```

Comment:
This might work for LARS but doesn't for other optimizers (see failing tests in the CI)

```python
FAILED tests/core/test_lightning_optimizer.py::test_lightning_optimizer[True]
FAILED tests/core/test_lightning_optimizer.py::test_lightning_optimizer[False]
FAILED tests/core/test_lightning_optimizer.py::test_params_groups_and_state_are_accessible
FAILED tests/trainer/test_trainer.py::test_trainer_access_in_configure_optimizers
FAILED tests/trainer/optimization/test_manual_optimization.py::test_step_with_optimizer_closure_with_different_frequencies
FAILED tests/trainer/optimization/test_optimizers.py::test_none_optimizer - V...
```